app/controllers/admin/lead_providers_controller.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+class Admin::LeadProvidersController < ApplicationController
+  def index
+    @lead_providers = LeadProvider.all
+  end
+
+  def new
+    @lead_provider = LeadProvider.new
+  end
+
+  def create
+    @lead_provider = LeadProvider.new(name: params.dig(:lead_provider, :name))
+
+    if @lead_provider.save
+      redirect_to admin_lead_providers_path
+    else
```

Comment:
I was thinking whether we could use `LeadProvider.create(...)` for something shorter here, but I guess the fact that we also want to assign value to `@lead_provider` when validation fails means we should have too much to put in an if statement anyway. 

Out of curiosity, does ruby let you do something as naughty as
```
if @lead_provider = LeadProvider.create(name: params.dig(:lead_provider, :name))
  redirect_to admin_lead_providers_path
else
  render :new
end
```
?

Reply:
I don't think that would work. From https://api.rubyonrails.org/classes/ActiveRecord/Persistence/ClassMethods.html#method-i-create:

> Creates an object (or multiple objects) and saves it to the database, if validations pass. The resulting object is returned whether the object was saved successfully to the database or not.

Reply:
I think something like 
`
@lead_provider = LeadProvider.create(name: params.dig(:lead_provider, :name))
if @lead_provider.valid?
  redirect_to admin_lead_providers_path
else
  render :new
end
`
might work, but I'm not 100%

Reply:
Pretty sure it would, but it wouldn't really be any nicer than the present implementation.

Reply:
Perhaps `@lead_provider.persisted?` would be better